src/NetAppFiles/NetAppFiles/Pool/UpdateNetAppFilesPool.cs

```diff
@@ -0,0 +1,133 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.NetAppFiles.Common;
+using Microsoft.Azure.Commands.NetAppFiles.Models;
+using Microsoft.Azure.Management.NetApp;
+using Microsoft.Azure.Management.NetApp.Models;
+
+namespace Microsoft.Azure.Commands.NetAppFiles.Pool
+{
+    [Cmdlet(
+        "Update",
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetAppFilesPool",
+        SupportsShouldProcess = true,
+        DefaultParameterSetName = FieldsParameterSet), OutputType(typeof(PSNetAppFilesPool))]
+    [Alias("Update-AnfPool")]
+    public class UpdateAzureRmNetAppFilesPool : AzureNetAppFilesCmdletBase
+    {
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The resource group of the ANF account")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter()]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The location of the resource")]
+        [ValidateNotNullOrEmpty]
+        [LocationCompleter("Microsoft.NetApp/netAppAccounts/capacityPools")]
+        public string Location { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The name of the ANF account")]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(
+            "Microsoft.NetApp/netAppAccounts",
+            nameof(ResourceGroupName))]
+        public string AccountName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The name of the ANF pool")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ParentObjectParameterSet,
+            HelpMessage = "The name of the ANF pool")]
+        [ValidateNotNullOrEmpty]
+        [Alias("PoolName")]
+        [ResourceNameCompleter(
+            "Microsoft.NetApp/netAppAccounts/capacityPools",
+            nameof(ResourceGroupName),
+            nameof(AccountName))]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
```

Comment:
@leonardbf for the property parameters (`-PoolSize` and `-ServiceLevel`), since this is a PATCH operation (since the verb of the cmdlet is `Update`), these parameters should be optional so that if the user doesn't provide a value for one of the parameters, its value on the server remains the same

Reply:
Yes there is a problem here. These values should be optional but actually they were omitted in the swagger for update. therefore patch is not really available here and the actual call is to CreateOrUpdate. This is deliberate and temporary. I opted to use Update rather than Set for minimum impact later. The new version of the swagger is already produced and I expect to be changing this PowerShell in a week or so to use the Update (when I started this the swagger was not yet done and the dotNet SDK isn't updated yet at this point but should be soon)